types/twitter_cldr/index.d.ts

```diff
@@ -0,0 +1,173 @@
+export type TimeUnit = 'second' | 'minute' | 'hour' | 'day' | 'week' | 'month' | 'year';
+export type PluralRule = 'zero' | 'one' | 'two' | 'few' | 'many' | 'other';
+export type RbnfGroupName = 'SpelloutRules' | 'OrdinalRules';
+
+export interface BidiString {
+    toString(): string;
+    reorder_visually(): BidiString;
+}
+
+export interface Currency {
+    country: string;
+    cldr_symbol: string;
+    symbol: string;
+    currency: string;
+    name: string;
+    code_points: number[];
+}
+
+export interface UnicodeRegex {
+    match(str: string): string[];
+    to_regexp(): RegExp;
+}
+
+export interface TwitterCldr {
+    Settings: {
+        locale(): Locale;
+    };
+
+    set_data(data: CldrData): null;
+    get_data(): CldrData;
+
+    DateTimeFormatter: {
+        new (): {
+            format(
+                date: Date,
+                options:
+                    | {
+                          format?: 'date' | 'time';
+                          type?: 'full' | 'long' | 'medium' | 'short';
+                      }
+                    | {
+                          format: 'additional';
+                          type: string;
+                      },
+            ): string;
+        };
+
+        additional_formats(): string[];
+    };
+
+    TimespanFormatter: {
+        new (): {
+            format(
+                timeSpanSeconds: number,
+                options?:
+                    | {
+                          unit?: TimeUnit;
+                          approximate?: boolean;
+                      }
+                    | {
+                          direction: 'none';
+                          unit?: TimeUnit;
+                          approximate?: boolean;
+                          type?: 'short' | 'abbreviated';
+                      },
+            ): string;
+        };
+    };
+
+    DecimalFormatter: {
+        new (): {
+            format(num: number, options?: { precision?: number }): string;
+        };
+    };
+
+    ShortDecimalFormatter: {
+        new (): {
+            format(num: number, options?: { precision?: number }): string;
+        };
+    };
+
+    LongDecimalFormatter: {
+        new (): {
+            format(num: number, options?: { precision?: number }): string;
+        };
+    };
+
+    CurrencyFormatter: {
+        new (): {
+            format(
+                num: number,
+                options?: {
+                    currency?: string;
+                    precision?: number;
+                },
+            ): string;
+        };
+    };
+
+    PercentFormatter: {
+        new (): {
+            format(num: number, options?: { precision?: number }): string;
+        };
+    };
+
+    Currencies: {
+        currency_codes(): string[];
+        for_code(currencyCode: string): Currency;
+    };
+
+    PluralRules: {
+        rule_for(num: number): PluralRule;
+        all(): PluralRule[];
+    };
+
+	/** Rule-Based Number Formatting */
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/twitter_cldr/index.d.ts;
  Line: 127;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.